infra/src/test/kotlin/fi/fta/geoviite/infra/common/GeoviiteOidDaoIT.kt

```diff
@@ -0,0 +1,58 @@
+package fi.fta.geoviite.infra.common
+
+import fi.fta.geoviite.infra.DBTestBase
+import fi.fta.geoviite.infra.tracklayout.LayoutKmPost
+import fi.fta.geoviite.infra.util.getOid
+import fi.fta.geoviite.infra.util.queryOne
+import kotlin.test.assertEquals
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+
+@ActiveProfiles("dev", "test")
+@SpringBootTest
+class GeoviiteOidDaoIT @Autowired constructor(val geoviiteOidDao: GeoviiteOidDao) : DBTestBase() {
+
+    @Test
+    fun `create a new oid sequence and generate oids from it`() {
+        transactionalForcingRollback {
+            jdbc.execute("""insert into common.oid_types values('abc')""") { it.execute() }
```

Comment:
Koska Kotlin-puolen OidType on vielä tyhjä, OidDaon ainoaa funktiota ei pysty testaamaan vielä lainkaan. Mutta tuleepahan testattua muuten migraation luomien juttujen toimivuus.

Reply:
Pakotin tässä rollbäkin siksi, koska osa testattavasta asiasta on tietokantatriggeri, joka käy tekemässä kantaan jotain omia muutoksiaan, ja jättäisin sen mieluummin yksityiskohdaksi, johon testi ei ota suoraan kantaa, mutta jota se ei myöskään halua jättää lojumaan kannan tilaan. Toinen vaihtoehto tietenkin olisi heittää triggeri ö-mappiin, mutta siihen en halua mennä, että testeissä olisi siivouslogiikkaa, missä pitää katsoa että mitäs sekvenssejä tässä nyt on luotu että saa ne poistettua.

Reply:
Totta, enpäs huomioinutkaan triggereitä kommenttia kirjoittaessani. Lienee siis oikeastaan hyvä juuri näin